docs/en/integrations/data-ingestion/apache-spark/index.md

```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
+        Dataset<Row> df1 = spark.read().jdbc(jdbcURL, String.format("(%s)", query), jdbcProperties);
+
+        df1.show();
+
+        //---------------------------------------------------------------------------------------------------
+        // Load the table from ClickHouse using load method
+        //---------------------------------------------------------------------------------------------------
+        Dataset<Row> df2 = spark.read()
+                .format("jdbc")
+                .option("url", jdbcURL)
+                .option("user", "default")
+                .option("password", "123456")
+                .option("query", query)
+                .load();
+
+
+        df2.show();
 
-        // Show the DataFrame
-        df.show();
 
         // Stop the Spark session
         spark.stop();
     }
 ```
+</TabItem>
+<TabItem value="Scala" label="Scala">
+
+```java
+object ReadData extends App {
+  // Initialize Spark session
+  val spark: SparkSession = SparkSession.builder.appName("example").master("local").getOrCreate
+
+  val jdbcURL = "jdbc:ch://localhost:8123/default"
+  val query: String = "select * from example_table where id > 2"
+
+
+  //---------------------------------------------------------------------------------------------------
+  // Load the table from ClickHouse using jdbc method
+  //---------------------------------------------------------------------------------------------------
+  val connectionProperties = new Properties()
+  connectionProperties.put("user", "default")
+  connectionProperties.put("password", "123456")
+
+  val df1: Dataset[Row] = spark.read.
+    jdbc(jdbcURL, s"($query)", connectionProperties)
+
+  df1.show()
+  //---------------------------------------------------------------------------------------------------
+  // Load the table from ClickHouse using load method
+  //---------------------------------------------------------------------------------------------------
+  val df2: Dataset[Row] = spark.read
+    .format("jdbc")
+    .option("url", jdbcURL)
+    .option("user", "default")
+    .option("password", "123456")
+    .option("query", query)
+    .load()
+
+  df2.show()
+
+
+
+  // Stop the Spark session// Stop the Spark session
+  spark.stop()
+
+}
+```
+
+Find more details on the [`s3` table function page](/docs/en/sql-reference/table-functions/s3.md).
+
+</TabItem>
+<TabItem value="Python" label="Python">
+
+```python
+from pyspark.sql import SparkSession
+
+jar_files = [
+    "jars/clickhouse-jdbc-X.X.X-SNAPSHOT-all.jar"
+]
+
+# Initialize Spark session with JARs
+spark = SparkSession.builder \
+    .appName("example") \
+    .master("local") \
+    .config("spark.jars", ",".join(jar_files)) \
+    .getOrCreate()
+
+url = "jdbc:ch://localhost:8123/default"
+user = "your_user" 
+password = "your_password"  
+query = "select * from example_table where id > 2"
+driver = "com.clickhouse.jdbc.ClickHouseDriver"
+
+df = (spark.read
```

Comment:
I think we should write meaningful variables names

Reply:
Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df